ui/docs/reference/core/worker-manager/google.md

```diff
@@ -0,0 +1,42 @@
+order: 10
+---
+import SchemaTable from 'taskcluster-ui/components/SchemaTable'
+
+# Google Provider Type
+
+Google-based providers create workers dynamically in GCE, using a single GCP project.
+Best practice is to use a dedicated project for each provider that is not shared with other uses.
+This provides additional security both for the workers and for any other GCP resources that the workers might otherwise accidentally gain access to.
+
+## Provider Configuration
+
+A google-based provider is be configured in `providers` with the following structure:
+
+```json
+{
+  "myProvider": {
+    "providerType": "google",
+    "project": "<gcp project identifier>",
+    "instancePermissions": "<instance permissions>",
+    "creds": "<google credentials>",
+    "credsFile": "<filename containing google credentials>"
+  },
+  ...
+}
+```
+
+The `project` configuration names the project in which instances will be created.
+It is the project *ID* and not the human-readable project name.
+
+The `instancePermissions` configuration defines the permissions granted to the GCP role assumed by the workers.
+TODO: what does this look like? does everyone need it?
+
+The GCP credentials are provided either in string form (`creds`) or in a file (`credsFile`).
+In either case, the data is the large JSON object containing a service account's keys.
+TODO: what permissions must this service account possess?
```

Comment:
Ah, of course! It needs the roles:

* `roles/iam.serviceAccountAdmin`
* `roles/iam.roleAdmin`
* `roles/resourcemanager.projectIamAdmin`
* `roles/compute.admin`

Note that these are very powerful so this should probably be a project dedicated to just these workers.

Reply:
Thanks!